db/db_impl.cc

```diff
@@ -1337,33 +1337,94 @@ std::vector<Status> DBImpl::MultiGet(
   struct MultiGetColumnFamilyData {
     ColumnFamilyData* cfd;
     SuperVersion* super_version;
+    MultiGetColumnFamilyData(ColumnFamilyData* cf, SuperVersion* sv)
+      : cfd(cf), super_version(sv) {}
   };
-  std::unordered_map<uint32_t, MultiGetColumnFamilyData*> multiget_cf_data;
-  // fill up and allocate outside of mutex
+  std::unordered_map<uint32_t, MultiGetColumnFamilyData> multiget_cf_data(column_family.size());
   for (auto cf : column_family) {
     auto cfh = reinterpret_cast<ColumnFamilyHandleImpl*>(cf);
     auto cfd = cfh->cfd();
     if (multiget_cf_data.find(cfd->GetID()) == multiget_cf_data.end()) {
-      auto mgcfd = new MultiGetColumnFamilyData();
-      mgcfd->cfd = cfd;
-      multiget_cf_data.insert({cfd->GetID(), mgcfd});
+      multiget_cf_data.emplace(cfd->GetID(), MultiGetColumnFamilyData(cfd, nullptr));
```

Comment:
I belive this should be `multiget_cf_data.emplace(cfd->GetID(), cfd, nullptr)`

Reply:
It doesn't work. It has to be something like ```multiget_cf_data.emplace(std::piecewise_construct, std::forward_as_tuple(cfd->GetID()), std::forward_as_typle(cfd, nullptr))```, but the current version looks simpler and not much overhead anyway.

Reply:
Forget about it then.